File: 66-PlusOne/66-PlusOne.java
// Last updated: 6/10/2026, 7:27:14 PM
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        int rem = digits[n-1] + 1 > 9 ? 1 : 0;
        digits[n-1] = (digits[n-1] + 1) % 10;
        for(int i = n-2 ; i>=0 ; i--) {
            int nr = digits[i] + rem > 9 ? 1 : 0;
            digits[i] = (digits[i] + rem) % 10;
            rem = nr;
        }
        if(rem == 1) {
            int[] ans = new int[n+1];
            ans[0] = 1;
            for(int i=0 ; i<n ; i++) ans[i+1] = digits[i];
            return ans;
        } else 
            return digits;
    }
}

File: 66-PlusOne/66-PlusOne.cpp
// Last updated: 6/10/2026, 7:27:19 PM
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> ans;
        int n = digits.size();
        int rem = 1;
        for(int i=n-1 ; i>=0 ; i--) {
            ans.push_back((rem + digits[i]) % 10);
            rem = (rem + digits[i]) / 10;
        }
        if(rem) ans.push_back(rem);
        reverse(ans.begin() , ans.end());
        return ans;
    }
};